simple_parsing/helpers/serialization/serializable.py

```diff
@@ -652,13 +654,24 @@ def to_dict(dc, dict_factory: type[dict] = dict, recurse: bool = True) -> dict:
             d[name] = custom_encoding_fn(value)
             continue
 
+        if add_selection:
+            ###### insert subgroups selected key
+            subgroups_dict = f.metadata.get("subgroups")
+            if subgroups_dict:
+                for g_name, g_cls in subgroups_dict.items():
+                    if isinstance(value, g_cls):
+                        _target = f"__subgroups__@{name}"
+                        d[_target] = g_name
```

Comment:
Nitpick: Reducing the nesting a bit:
```suggestion
        if add_selection and "subgroups" in f.metadata:
            ###### insert subgroups selected key
            subgroups_dict = f.metadata["subgroups"]
            for g_name, g_cls in subgroups_dict.items():
                if isinstance(value, g_cls):
                    _target = f"__subgroups__@{name}"
                    d[_target] = g_name
```

Also, why do we care about the chosen subgroup name here? Can't we just store the class to use?

Reply:
Ah, we store the subgroup names, because the subgroup class to use would be harder to serialize. That makes sense.